File: backend/src/main/java/iuh/fit/backend/model/Review.java
package iuh.fit.backend.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

// Review Entity
@Entity
@Table(name = "reviews")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Review {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "user_id")
    private Integer userId;

    @Column(name = "product_id")
    private Integer productId;

    private String content;
    private Integer rating;
    private String title;
    private String email;
    private String nickname;

    @Column(name = "isRecommend")
    private Boolean isRecommend;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @ManyToOne
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    @JsonIgnore
    private User user;

    @ManyToOne
    @JoinColumn(name = "product_id", insertable = false, updatable = false)
    @JsonIgnore
    private Product product;

    @OneToMany(mappedBy = "review")
    @JsonIgnore
    private List<ReviewImage> reviewImages;
}

File: frontend/src/types/Review.ts
import type { User } from './User';
import type { Product } from './Product';
import type { ReviewImage } from './ReviewImage';

export interface Review {
  id: number;
  userId: number;
  productId: number;
  content: string;
  rating: number;
  title: string;
  email: string;
  nickname: string;
  isRecommend: boolean;
  createdAt: string;
  updatedAt: string;
  user?: User;
  product?: Product;
  reviewImages?: ReviewImage[];
}
